sway-core/src/control_flow_analysis/dead_code_analysis.rs

```diff
@@ -148,53 +148,70 @@ impl ControlFlowGraph {
+                            ControlFlowGraphNode::ProgramNode(TypedAstNode {
+                                content:
+                                    TypedAstNodeContent::Declaration(
+                                        TypedDeclaration::FunctionDeclaration(
+                                            TypedFunctionDeclaration {
+                                                visibility: Visibility::Public,
+                                                ..
+                                            },
+                                        ),
+                                    ),
+                                ..
+                            }) => true,
+                            ControlFlowGraphNode::ProgramNode(TypedAstNode {
+                                content:
+                                    TypedAstNodeContent::Declaration(
+                                        TypedDeclaration::TraitDeclaration(TypedTraitDeclaration {
+                                            visibility: Visibility::Public,
+                                            ..
+                                        }),
+                                    ),
+                                ..
+                            }) => true,
+                            ControlFlowGraphNode::ProgramNode(TypedAstNode {
+                                content:
+                                    TypedAstNodeContent::Declaration(
+                                        TypedDeclaration::StructDeclaration(
+                                            TypedStructDeclaration {
+                                                visibility: Visibility::Public,
+                                                ..
+                                            },
+                                        ),
+                                    ),
+                                ..
+                            }) => true,
+                            ControlFlowGraphNode::ProgramNode(TypedAstNode {
+                                content:
+                                    TypedAstNodeContent::Declaration(TypedDeclaration::ImplTrait {
+                                        ..
+                                    }),
+                                ..
+                            }) => true,
+                            // Don't emit warnings for constant declarations in libraries. We
+                            // cannot yet figure out whether they're actually used outside the
+                            // library or not.
+                            ControlFlowGraphNode::ProgramNode(TypedAstNode {
+                                content:
+                                    TypedAstNodeContent::Declaration(
+                                        TypedDeclaration::ConstantDeclaration(_),
+                                    ),
+                                ..
+                            }) => true,
```

Comment:
This is the only actual addition to this file. Everything else is due to the formatter.